app/models/session_activation.rb

```diff
@@ -53,4 +59,22 @@ def exclusive(id)
       where('session_id != ?', id).destroy_all
     end
   end
+
+  private
+
+  def assign_user_agent
+    self.user_agent = '' if user_agent.nil?
+  end
+
+  def assign_access_token
+    superapp = Doorkeeper::Application.find_by(superapp: true)
+
+    return if superapp.nil?
```

Comment:
This must have existed previously. The old code was in `HomeController#find_or_create_access_token`, which also called `Doorkeeper::Application.where(superapp: true).first` - how could those instances have worked then? Unless nil was allowed for application_id on oauth_access_tokens, which I didn't check.